packages/unmock-core/src/service/interfaces.ts

```diff
@@ -17,21 +23,29 @@ const RESTMethodTypes = [
   "delete",
   "options",
   "trace",
-  "all", // internally used to mark all the above methods.
 ] as const;
+
+const DEF_REST_METHOD = [DEFAULT_HTTP_METHOD] as const;
+
+type DEFAULT_HTTP_METHOD_AS_TYPE = typeof DEF_REST_METHOD[number];
 export type HTTPMethod = typeof RESTMethodTypes[number];
+export type ExtendedHTTPMethod = HTTPMethod | DEFAULT_HTTP_METHOD_AS_TYPE;
 
 export const isRESTMethod = (maybeMethod: string): maybeMethod is HTTPMethod =>
   RESTMethodTypes.toString().includes(maybeMethod.toLowerCase());
+export const isExtendedRESTMethod = (
+  maybeMethod: string,
+): maybeMethod is ExtendedHTTPMethod =>
+  maybeMethod === DEFAULT_HTTP_METHOD || isRESTMethod(maybeMethod);
 
 export interface IServiceMapping {
   [serviceName: string]: IService;
 }
 
 export interface IStateInput {
-  method: HTTPMethod;
+  method: ExtendedHTTPMethod;
   endpoint: string;
-  newState: IUnmockServiceState;
+  newState: UnmockServiceState;
```

Comment:
Moving all the state stuff (in future) to a directory of their own would also shorten this file quite a bit.